hip/matrix/dense_kernels.hip.cpp

```diff
@@ -311,25 +300,14 @@ void convert_to_csr(std::shared_ptr<const HipExecutor> exec,
                        stride, as_hip_type(source->get_const_values()),
                        as_hip_type(row_ptrs));
 
-    size_type grid_dim = ceildiv(num_rows + 1, default_block_size);
-    auto add_values = Array<IndexType>(exec, grid_dim);
+    prefix_sum(exec, row_ptrs, num_rows + 1);
 
-    hipLaunchKernelGGL(HIP_KERNEL_NAME(start_prefix_sum<default_block_size>),
-                       dim3(grid_dim), dim3(default_block_size), 0, 0,
-                       num_rows + 1, as_hip_type(row_ptrs),
-                       as_hip_type(add_values.get_data()));
-
-    hipLaunchKernelGGL(HIP_KERNEL_NAME(finalize_prefix_sum<default_block_size>),
-                       dim3(grid_dim), dim3(default_block_size), 0, 0,
-                       num_rows + 1, as_hip_type(row_ptrs),
-                       as_hip_type(add_values.get_const_data()));
+    size_type grid_dim = ceildiv(num_rows, default_block_size);
```

Comment:
I guess this has been updated to reflect that the `fill_in_csr` needs only grid dimensions `num_rows/{denom}` instead of `num_rows+1/{denom}` as was present before ?

Reply:
Yes, in some rare cases, we would be launching an unnecessary thread block, but I introduced that change mostly to avoid confusion.